### PR TITLE
Ignore HTML comments

### PR DESCRIPTION
**Pull request by mmacia**
Hello, I found that when I try to render nested components I got this kind of errors in my browser console.

Screenshot_20240307_174120

This happens because each component is surrounded by an HTML comment. In this PR I just skip the comment nodes I found.

The error disappears in my project and everything is working properly but let me know if I have to take into account anything else.

### CODE REVIEW

**Comment by chrismccord** [comment]
The heex compiler also requires an HTML tag at the root of a livecomponent so your project should have failed to compile with leading comments in an LiveComponent. Thanks!

**Comment by SteffenDE** [comment]
@chrismccord I think you misunderstood the issue. The warning is caused by the heex debug annotations when they render in a phx-update append / prepend if I understood this correctly.

**Comment by chrismccord** [comment]
ahhh, the debug annotations

File: assets/js/phoenix_live_view/dom.js [hunk 1]
```diff
@@ -519,7 +519,7 @@ let DOM = {
         if(!childNode.id){
           // Skip warning if it's an empty text node (e.g. a new-line)
           let isEmptyTextNode = childNode.nodeType === Node.TEXT_NODE && childNode.nodeValue.trim() === ""
-          if(!isEmptyTextNode){
+          if(!isEmptyTextNode && && childNode.nodeType !== childNode.COMMENT_NODE){
```
Comment (chrismccord) [inline assets/js/phoenix_live_view/dom.js:522]:
```suggestion
          if(!isEmptyTextNode && childNode.nodeType !== Node.COMMENT_NODE){
```

**Comment by chrismccord** [comment]
❤️❤️❤️🐥🔥

**Pull request merged**